IRescue/UserLocalisation/Sensors/Marker/MarkerSensor.cs

```diff
@@ -20,21 +20,31 @@ public class MarkerSensor : IPositionSource, IOrientationSource
         private readonly float standardDeviation;
 
         /// <summary>
-        ///   The predicted locations.
+        ///  The MarkerLocations.
         /// </summary>
-        private List<Measurement<Vector3>> positions;
+        private MarkerLocations markerLocations;
 
         /// <summary>
         ///   The predicted rotations.
         /// </summary>
-        private List<Measurement<Vector3>> orientations;
+        private Measurement<Vector3>[] orientations;
 
         /// <summary>
-        ///  The MarkerLocations.
+        ///   The predicted locations.
         /// </summary>
-        private MarkerLocations markerLocations;
+        private Measurement<Vector3>[] positions;
 
         /// <summary>
+        /// Current index pointer;
+        /// </summary>
+        private int pointer = 0;
+
+        /// <summary>
+        /// Length of the buffer.
+        /// </summary>
+        private int bufferLength = 30;
```

Comment:
I think this shoud be set with a parameter in the constructor instead of being hardcoded. In the IMUSource we do it that way as well.